TaskHouseUnitTests/WorkersControllerUnitTests.cs

```diff
@@ -0,0 +1,207 @@
+        }
+
+        [Fact]
+        public async void WorekrController_Create_ReturnsBadRequest_WhenGivenNullWorker()
+        {
+            Worker worker = null;
+
+            var result = await controller.Create(worker);
+
+            Assert.IsType<BadRequestObjectResult>(result);
+
+
+        }
+
+        [Fact]
+        public async void WorkerController_Update_ReturnsBadRequestResult_WhenIdIsInvalidAndWorkerIsNull()
+        {
+            Worker worker = null;
+            int id = 2600;
+
+            var result = await controller.Update(id, worker);
+
+            Assert.IsType<BadRequestResult>(result);
+        }
+
+        [Fact]
+        public async void WorkerController_Update_ReturnsBadRequestResult_WhenIdIsInvalid()
+        {
+            Worker TestWorker = new Worker()
+            {
+                Id = 1,
+                Username = "Tusername2",
+                Password = "+z490sXHo5u0qsSaxbBqEk9KsJtGqNhD8I8mVBdDJls=", //1234
+                Email = "test@test.com",
+                FirstName = "Bob7",
+                LastName = "Bobsen6",
+                Salt = "upYKQSsrlub5JAID61/6pA=="
+            };
+
+            int id = 10000000;
+
+            var result = await controller.Update(id, TestWorker);
+
+            Assert.IsType<BadRequestResult>(result);
+        }
+
+        [Fact]
+        public async void WorkerController_Delete_ReturnsNoContentResult_WhenIdIsValid()
+        {
+            int Id = 1;
+            var result = await controller.Delete(Id);
+
+            Assert.IsType<NoContentResult>(result);
+        }
+
+        [Fact]
+        public async void WorkerController_Update_ReturnsBadRequestResult_WhenWorkerIsNull()
+        {
+            Worker worker = null;
+            int id = 1;
+
+            var result = await controller.Update(id, worker);
+
+            Assert.IsType<BadRequestResult>(result);
+        }
+
+        [Fact]
+        public async void WorkerController_Delete_ActuallyDeletes_WhenIdIsValid()
+        {
+            Worker TestWorker = new Worker()
+            {
+                Id = 5,
+                Username = "Tusername",
+                Password = "+z490sXHo5u0qsSaxbBqEk9KsJtGqNhD8I8mVBdDJls=", //1234
+                Email = "test@test.com",
+                FirstName = "Bob7",
+                LastName = "Bobsen6",
+                Salt = "upYKQSsrlub5JAID61/6pA=="
+            };
+
+            await controller.Delete(TestWorker.Id);
+            var result = await controller.Get(TestWorker.Id);
+
+            Assert.IsType<NotFoundResult>(result);
+        }
+
+        [Fact]
+        public async void WorkerController_Delete_ReturnsNotFoundResult_WhenIdIsInvalid()
+        {
+            int id = 2500;
+      
+            var result = await controller.Delete(id);
+
+            Assert.IsType<NotFoundResult>(result);
+        }
+
+        [Fact]
+        public async void WorkerController_Update_ReturnsNoContentResult_withValidIdAndValidWorker()
```

Comment:
The name of the method doesn't make sense it returns a result, so the `ReturnsNoContentResult` part is not valid. This is a problem in some of your test cases. Please ensure that the method names actually describes what the tests purpose is. This will increase maintainability and will make debugging easier when a test fails.